lib/eco/treelexer/lexer.py

```diff
@@ -229,13 +233,24 @@ def __init__(self, rules):
             self.patterns.append((pattern, name))
 
     def lex(self, text):
+        """Lexes a given string by trying all patterns. When a match is found a
+        token is created, and the lexing continues with the remainder of the
+        string. Lookahead is calculated by observing how far the patternmatcher
+        got before returning 'no match'. If the pattern is open ended, e.g. +,
+        *, etc, the match is not exact and we need to add 1 to the lookahead,
```

Comment:
Not sure what "need to add 1" means... Is it a fixed thing or infinite or ...?

Reply:
If the token was matched by `+` or `*` then it always has a lookahead of at least 1. If it already has a lookahead > 1 then it's lookahead is 1 more as it needs to look at 1 additional character to know that there is no longer match.

Reply:
OK. Maybe a quick rephrasing might help here?